src/neighborhoods/metricball.jl

```diff
@@ -36,40 +36,29 @@ struct MetricBall{R,M} <: Neighborhood
   metric::M
 end
 
-function MetricBall(radii::SVector{Dim,T}, rotation=nothing) where {Dim,T}
-  # default rotation
-  rot = if isnothing(rotation)
-    if Dim == 2
-      ClockwiseAngle(zero(T))
-    elseif Dim == 3
-      EulerAngles(zeros(T, Dim)...)
-    else
-      throw(ErrorException("not implemented"))
-    end
-  else
-    rotation
-  end
+function _default_rotation(::Val{2}, T)
+  one(Angle2d{T})
+end
+
+function _default_rotation(::Val{3}, T)
+  one(QuatRotation{T})
+end
 
+function MetricBall(radii::SVector{Dim,T}, R::Rotation{Dim}=_default_rotation(Val{Dim}(), T)) where {Dim,T}
   # scaling matrix
   Λ = Diagonal(one(T) ./ radii .^ 2)
 
-  # rotation matrix
-  R = convert(DCM, rot)
-
-  # sanity check
-  @assert size(R) == (Dim, Dim) "invalid rotation for radii"
-
   # Mahalanobis metric
   metric = Mahalanobis(Symmetric(R'*Λ*R))
 
   MetricBall{typeof(radii),typeof(metric)}(radii, metric)
 end
 
-MetricBall(radii::NTuple{Dim,T}, rotation=nothing) where {Dim,T} =
+MetricBall(radii::NTuple{Dim,T}, rotation=_default_rotation(Val{Dim}(), T)) where {Dim,T} =
   MetricBall(SVector(radii), rotation)
 
 # avoid silent calls to inner constructor
-MetricBall(radii::AbstractVector{T}, rotation=nothing) where {T} =
+MetricBall(radii::AbstractVector{T}, rotation=_default_rotation(Val{Dim}(), T)) where {T} =
```

Comment:
```suggestion
MetricBall(radii::AbstractVector{T}, rotation=default_rotation(Val{Dim}(), T)) where {T} =
```

Reply:
The usage of `_default_rotation` is totally internal. so I have named it with the prefix `_` to represent that.

[Prepended underscore in Base functions](https://discourse.julialang.org/t/prepended-underscore-in-base-functions/19645/3):

>In general, an prepended underscore is typically used to indicate some kind of internal implementation detail, especially functions that make special assumptions about the arguments and so aren’t safe for external use.

Are there any conventions to avoid these underscores in Meshes.jl?

Reply:
We avoid underscores as much as possible in Meshes.jl, that is why I suggested removing it. We recently merged a formal code style with JuliaFormatter, so these things will gradually converge.

Reply:
BTW, we need to rebase this PR on top of the latest master to fix these formatting conflicts.

Reply:
We can simply modify the internal functions, they are not exported and not intended for end-users.